src/prog_models/prognostics_model.py

```diff
@@ -404,6 +404,30 @@ def next_state(self, x, u, dt) -> dict:
         dx = self.dx(x, u)
         return {key: x[key] + dx[key]*dt for key in dx.keys()}
 
+    def apply_limits(self, x):
+        """
+        Apply state bound limits. Any state outside of limits will be set to the closest limit.
+
+        Parameters
+        ----------
+        x : dict
+            state, with keys defined by model.states \n
+            e.g., x = {'abc': 332.1, 'def': 221.003} given states = ['abc', 'def']
+
+        Returns
+        -------
+        x : dict
+            Bounded state, with keys defined by model.states
+            e.g., x = {'abc': 332.1, 'def': 221.003} given states = ['abc', 'def']
+        """
+        for (key, limit) in self.state_limits.items():
```

Comment:
This is a user-defined function, right? If the key doesn't exist in the state, does the user get the Python "KeyError"? is that ok for your users?

Reply:
A key error should be clear enough. Also, since this is in the innermost loop, I don't want to do my own check and return my own custom error message. 

In the future there might be a way for me to check each time state_limits is updated. I'll open an issue to look at that.

Reply:
Oh, I just realized I missed a question:

> This is a user-defined function, right?
state_limits is a user-defined python dictionary, and items() is a function available in any dictionary. 

To be properly 'pythonic' I don't check that it's a dictionary. This allows users to create their own class to replace it as desired (as long as it's utterable and has the function items() that returns keys and limits.